istioctl/pkg/verifier/verifier.go

```diff
@@ -0,0 +1,343 @@
+			fixTimestampRelatedUnmarshalIssues(un)
+			by := util.ToYAML(un)
+			iop, err := operator_istio.UnmarshalIstioOperator(by, true)
+			if err != nil {
+				return err
+			}
+			if v.manifestsPath != "" {
+				iop.Spec.InstallPackagePath = v.manifestsPath
+			}
+			generatedCrds, generatedDeployments, err := v.verifyPostInstallIstioOperator(iop, filename)
+			if err != nil {
+				return err
+			}
+			crdCount += generatedCrds
+			istioDeploymentCount += generatedDeployments
+		default:
+			result := info.Client.
+				Get().
+				Resource(kinds).
+				Name(name).
+				Do(context.TODO())
+			if result.Error() != nil {
+				result = info.Client.
+					Get().
+					Resource(kinds).
+					Namespace(namespace).
+					Name(name).
+					Do(context.TODO())
+				if result.Error() != nil {
+					return istioVerificationFailureError(filename,
+						fmt.Errorf("the required %s:%s is not ready due to: %v",
+							kind, name, result.Error()))
+				}
+			}
+			if kind == "CustomResourceDefinition" {
+				crdCount++
+			}
+		}
+		v.logger.LogAndPrintf("%s: %s.%s checked successfully", kind, name, namespace)
+		return nil
+	})
+	return crdCount, istioDeploymentCount, err
+}
+
+// Find an IstioOperator matching revision in the cluster.  The IstioOperators
+// don't have a label for their revision, so we parse them and check .Spec.Revision
+func (v *StatusBasedVerifier) operatorFromCluster(revision string) (*v1alpha1.IstioOperator, error) {
+	restConfig, err := v.k8sConfig().ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	client, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	iops, err := AllOperatorsInCluster(client)
+	if err != nil {
+		return nil, err
+	}
+	for _, iop := range iops {
+		if iop.Spec.Revision == revision {
+			return iop, nil
+		}
+	}
+	return nil, fmt.Errorf("control plane revision %q not found", revision)
+}
+
+func (v *StatusBasedVerifier) reportStatus(crdCount, istioDeploymentCount int, err error) error {
+	if err != nil {
+		return err
+	}
+	v.logger.LogAndPrintf("Checked %v custom resource definitions", crdCount)
+	v.logger.LogAndPrintf("Checked %v Istio Deployments", istioDeploymentCount)
+	if istioDeploymentCount == 0 {
+		v.logger.LogAndPrintf("No Istio installation found")
+		return fmt.Errorf("no Istio installation found")
+	}
+	v.logger.LogAndPrintf("Istio is installed successfully")
+	return nil
+}
+
+func fixTimestampRelatedUnmarshalIssues(un *unstructured.Unstructured) {
+	un.SetCreationTimestamp(meta_v1.Time{}) // UnmarshalIstioOperator chokes on these
+
+	// UnmarshalIstioOperator fails because managedFields could contain time
+	// and gogo/protobuf/jsonpb(v1.3.1) tries to unmarshal it as struct (the type
+	// meta_v1.Time is really a struct) and fails.
+	un.SetManagedFields([]meta_v1.ManagedFieldsEntry{})
+}
+
+// Find all IstioOperator in the cluster.
+func AllOperatorsInCluster(client dynamic.Interface) ([]*v1alpha1.IstioOperator, error) {
```

Comment:
This func looks similar to [allOperatorsInCluster](https://github.com/istio/istio/blob/bfa65c6520b93a56db8b56df80e8e7bc29bcf4d6/istioctl/pkg/install/verify.go#L438) can you re-use it or delete that duplicate func?

Reply:
Actually, it is the same function :) I renamed it because it is also required in `pre-verify`.